File: 250902/1차원 젠가/jenga-1d.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.stream.IntStream;
// import java.util.ArrayList;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] blocks = new int[n];
        for (int i = 0; i < n; i++) {
            blocks[i] = sc.nextInt();
        }
        int s1 = sc.nextInt();
        int e1 = sc.nextInt();
        int s2 = sc.nextInt();
        int e2 = sc.nextInt();
        // Please write your code here.
        for (int i = 0; i<2; i++){
            if(i == 0){
                int[] t1 = Arrays.copyOfRange(blocks, 0, s1-1);
                int[] t2 = Arrays.copyOfRange(blocks, e1, blocks.length);
                blocks = IntStream.concat(Arrays.stream(t1), Arrays.stream(t2)).toArray();
            }else{
                int[] t1 = Arrays.copyOfRange(blocks, 0, s2-1);
                int[] t2 = Arrays.copyOfRange(blocks, e2, blocks.length);
                blocks = IntStream.concat(Arrays.stream(t1), Arrays.stream(t2)).toArray();
            }
        }
        System.out.println(blocks.length);
        Arrays.stream(blocks)
                .forEach(System.out::println);
    }
}

File: 250902/1차원 젠가/jenga-1d.py
n = int(input())
blocks = [int(input()) for _ in range(n)]
s1, e1 = map(int, input().split())
s2, e2 = map(int, input().split())

# Please write your code here.
for i in range(2):
    if i == 0:
        temp = blocks[0:s1-1] + blocks[e1:]
        blocks = temp
        # print(blocks)
    else:
        temp = blocks[0:s2-1] + blocks[e2:]
        blocks = temp
        # print(blocks)

print(len(blocks))
for i in blocks:
     print(i)